### PR TITLE
Travis-CI continuous integration

### PR DESCRIPTION
**Pull request by isuruf**
If you enable travis-ci for this repo, each push to branches and each PR will be tested and a status will be shown, so that you can see it passes the tests.

Two tests are added one for linux and one for osx. In OSX, `make check` is failing so I've made it such that `make check` will not fail the build.

### CODE REVIEW

**Comment by JohnCremona** [comment]
OK, I have activated TravisCI.

**Comment by JohnCremona** [comment]
... and have completed a successful build there (my first).  Did you keep a record of what failed on osx?

**Comment by isuruf** [comment]
See the log here, https://travis-ci.org/JohnCremona/eclib/jobs/271458971#L1146-L1204
On OSX, I set it up so that only a build failure would fail the build, since running `make check` will fail all the time.

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    - chmod +x autogen.sh
+    - ./autogen.sh
+
+    - chmod +x configure
+    - ./configure --prefix="$PREFIX" --with-ntl="$PREFIX" --with-pari="$PREFIX" --with-flint="$PREFIX" --with-boost="no" --disable-allprogs
+
+    - make
+    - |
+      if [[ "${TRAVIS_OS_NAME}" == "osx" ]]; then
+          set +e;
```
Comment (isuruf) [inline .travis.yml]: If you remove this line, OSX `make check` will fail

**Comment by JohnCremona** [comment]
Thanks -- I just found the log.  It is only numerical noise

**Pull request merged**